enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,306 @@
+  //
+  // Valid values are:
+  //
+  // * aws
+  // * azure
+  // * gcp
+  // * vSphere
+  //
+  // ---
+  // Consumers should treat unknown values as a NO-OP.
+  //
+  // +kubebuilder:validation:Required
+  // +unionDiscriminator
+  DriverType CSIDriverType `json:"driverType"`
+
+  // AWS is used to configure the AWS CSI driver.
+  // +optional
+  AWS *AWSCSIDriverConfigSpec `json:"aws,omitempty"`
+
+  // Azure is used to configure the Azure CSI driver.
+  // +optional
+  Azure *AzureCSIDriverConfigSpec `json:"azure,omitempty"`
+
+  // GCP is used to configure the GCP CSI driver.
+  // +optional
+  GCP *GCPCSIDriverConfigSpec `json:"gcp,omitempty"`
+
+  ... exisiting vsphere field ... 
+}
+```
+
+ and fields for the encryption keys will be added
+to platform-specific config specs:
+
+#### AWS
+
+```go
+// AWSCSIDriverConfigSpec defines properties that can be configured for the AWS CSI driver.
+type AWSCSIDriverConfigSpec struct {
+    // KMSKeyARN sets the cluster default storage class to encrypt volumes with a user-defined KMS key,
+    // rather than the default KMS key used by AWS.
+    // According to https://docs.aws.amazon.com/service-authorization/latest/reference/list_awskeymanagementservice.html
+    // the format for a KMS key ARN is:
+    // arn:${Partition}:kms:${Region}:${Account}:key/${KeyId}
+    // +optional
+    KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+}
+```
+#### Azure
+
+```go
+// AzureDiskEncryptionSet defines the configuration for a disk encryption set.
+type AzureDiskEncryptionSet struct {
+  // SubscriptionID defines the Azure subscription that contains the disk encryption set.
+  // When omitted, the subscription from the authenticating credentials of the CSI driver
+  // will be used.
+  SubscriptionID string `json:"subscriptionId"`
+  
+  // ResourceGroup defines the Azure resource group that contains the disk encryption set.
+  // When omitted, the cluster resource group from the cluster infrastructure object
+  // will be used.
+  ResourceGroup string `json:"resourceGroup"`
+  
+  // Name is the name of the disk encryption set that will be set on the default storage class.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength:=80
+  // +kubebuilder:validation:Pattern:=[a-zA-Z0-9_-]
+  Name string `json:"name"`
+}
+
+// AzureCSIDriverConfigSpec defines properties that can be configured for the Azure CSI driver.
+type AzureCSIDriverConfigSpec struct{
+  // DiskEncryptionSet sets the cluster default storage class to encrypt volumes with a
+  // customer-managed encryption set, rather than the default platform-managed keys.
+  // +optional
+  DiskEncryptionSet *AzureDiskEncryptionSet `json:"diskEncryptionSet,omitempty"`
+}
+
+```
+
+#### GCP
+
+```go
+// GCPKMSKeyReference gathers required fields for looking up a GCP KMS Key
+type GCPKMSKeyReference struct {
+  // Name is the name of the customer-managed encryption key to be used for disk encryption.
```

Comment:
Please explain the validations in the godoc so that they end up in the generated docs